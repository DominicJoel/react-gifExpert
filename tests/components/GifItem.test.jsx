import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Test of GifItem', () => {
      
  const title = 'Saitama';
  const  url = 'https://onepunch.com/saitama.jpg';

  test('should match the snapshot', () => {


    const {container} = render(<GifItem title={title} url={url} />)
    
    expect(container).toMatchSnapshot();

  });

  test('should show the correct img and alt', () => {
    
    const {container} = render(<GifItem title={title} url={url} />);
     
   // screen.debug(); // To show everything rendered


   // expect(screen.getByRole('img').src).toBe(url); // Esto se puede hacer con getById etc
    //expect(screen.getByRole('img').alt).toBe(title); // Esto se puede hacer con getById etc


    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  
  test('should show the test of the document', () => {
    const {container} = render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  })
  
})

// u - to update the snaphot