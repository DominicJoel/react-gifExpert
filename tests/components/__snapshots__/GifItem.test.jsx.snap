// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://onepunch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
